### PR TITLE
Add an overview and vision statement to the documentation landing page

### PR DESCRIPTION
**Pull request by tfoote**
This can help people understand the overall project.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -32,3 +32,32 @@ Welcome to maliput documentation!
...
+
+We expect to see adoption of Maliput, for example in programmatic simulated world generation.
+Projects such as CARLA are generating simulated testing worlds based on road networks.
+Higher-fidelity road networks can be fed into the generator to create comprehensive simulated worlds which match real world environments, with controlled variations for better coverage.
+
+Maliput has been guided by the needs of the TRI/Woven autonomous driving team and automotive use cases in general, but it also has wider applications.
+Maliput’s standard interface can be leveraged in almost every case where a collection of mobile robots are interacting in the same space and need to be deconflicted.
+We foresee users for Maliput in warehouses, port drayage, and airport aprons.
+Any system managing this sort of multiple robot interaction needs the kind of capability provided by Maliput.
+Existing fleet systems have leveraged some of these capabilities but in ad hoc and duplicative ways.
```
Comment (francocipollone) [inline docs/index.rst:63]: Let me suggest the following, not regarding content but doc organization:

1. I would add this info before the table of contents in a way that this is the first thing that you see.
2. I would move this text to a .rst file and from here include that file, so as to keep the `index.rst` file as simple as possible.
   ```rst
     .. include:: maliput_vision.rst
     
   ```

**Review by francocipollone — DISMISSED** [review]
lgtm, small comment.

File: docs/index.rst (hunk 1 above)
Comment (francocipollone) [reply docs/index.rst]: Check [6396460](https://github.com/maliput/maliput_documentation/pull/103/commits/63964608b1902fdce6d08044d470813b4436de7d)

**Comment by francocipollone** [comment]
There is something that makes some noise I believe:
(I generated locally that's why you don't see the readthedocs theme)
image

This index page contains a `Maliput Overview` section, however, there is also a `Maliput Overview` page for the overview over the library: https://maliput.readthedocs.io/en/latest/maliput_overview.html

We should rename either one of them probably to avoid any confusion.

**Comment by tfoote** [comment]
Thanks for the improved integration with the getting started content. The new layout looks good to me.

**Pull request merged**